pandas/core/groupby.py

```diff
@@ -2850,7 +2850,15 @@ def _get_grouper(obj, key=None, axis=0, level=None, sort=True,
     elif isinstance(key, BaseGrouper):
         return key, [], obj
 
-    # Everything which is not a list is a key (including tuples):
+    tuple_as_list = isinstance(key, tuple) and key not in obj
```

Comment:
The only disadvantage I see with this approach is that
```python
pd.DataFrame(1, index=range(3), columns=pd.MultiIndex.from_product([[1, 2], [3,4]])).groupby((7, 8)).mean()
```
will raise ``KeyError: 7`` while  ``KeyError: (7,8)`` would be more correct. Do you think

```python
isinstance(key, tuple) and key not in obj and set(key).issubset(obj)
```
is too expensive?

Reply:
this change would be ok

Reply:
(Except that, as @jorisvandenbossche reminded below, ``set(key)`` could contains non-hashable objects, so this possibility should be catched)